### DISCUSSION TITLE
Show a table with the versions of all package managers used for a run

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#30497] [Suggest an Idea]
### Intro

A user asked why Renovate removes the `license` field from `npm` package updates. It turned out that is a bug with `npm`, not with Renovate. I think npm version `10.8.0` is the first version that fixes the bug:

- https://github.com/renovatebot/renovate/discussions/30473#discussioncomment-10193823

But then the question is: "What version of `npm` did Renovate use this run?".

Feature idea: show table with the versions of all package managers used for a run.

### Interactive mockup

<details><summary>Select me to see the versions of package managers Renovate used this run</summary>
<p>

| Renovate package manager | Version |
| --- | --- |
| `npm` | `1.0.0` |
| `dockerfile` | `1.1.0` |
| `cake`| `1.3.0` |

</p>
</details> 

#### Code for mockup

```markdown
<details><summary>Select me to see the versions of package managers Renovate used this run</summary>
<p>

| Renovate package manager | Version |
| --- | --- |
| `npm` | `1.0.0` |
| `dockerfile` | `1.1.0` |
| `cake`| `1.3.0` |

</p>
</details> 
```

**Comment by rarkins** [comment]
I think in 99% of cases this won't be important, and not worth adding noise to our PRs. Instead it could be better if we add an intentional "version" command (e.g. `npm --version`) as part of each artifacts update so that it's included in the logs if there's a problem)=

**Reply by HonkingGoose** [reply]
Two questions, before I make an issue for us. :wink:

Should Renovate:

1. Print the "package manager version" _once_ per manager, or _always_?
2. Print the output to `INFO` or `DEBUG` log level?

| Print version | Log size | Runtime (for fetching versions) | Notes |
| :--- | :--- | :--- | :--- |
| Once | Smaller | Shorter | Edge-case: what if Renovate starts with package manager version `1`, but uses `2` on the next item for some reason? (Constraints, or other config?) |
| Always | Bigger | Longer | |

**Reply by rarkins** [reply]
It should be a part of the regular debug logging of child process stdout

**Reply by HonkingGoose** [reply]
Renovate has 90+ package managers. Most of them will have a `version` or `--version` command. How do you want to track this work?

- 1 issue, with lots of todos.
- Or, 90 small issues.

**Reply by rarkins** [reply]
Only a subset of those update artifacts though. I think one issue is best

**Reply by HonkingGoose** [reply]
Do you have a list of package managers that update artifacts for me?

I'll make a table based on that list. Then it's up to you, or someone else, to update the code. :wink: Example:

| Manager | Command to print version information | Notes |
| ---- | --- | --- |
| `npm` | `npm --version` | |

**Reply by rarkins** [reply]
In the source code somewhere there should be a list of all the containerbase tools we support, which is essentially the same thing

**Reply by HonkingGoose** [reply]
Is this what you mean?

https://github.com/renovatebot/renovate/blob/833df8e32ba2fc0a0e3b835b0a0face42a6cb49c/lib/util/exec/containerbase.ts#L19-L214

**Comment by HonkingGoose** [comment]
See issue:

- #30824
